src/plugins/discover/public/application/main/discover_main_route.tsx

```diff
@@ -115,6 +117,11 @@ export function DiscoverMainRoute({
       if (savedSearchId) {
         return true; // bypass NoData screen
       }
+
+      if (isOfEsqlQueryType(stateContainer.appState.getState().query)) {
```

Comment:
@clintandrewhall - is this check here in Discover to account for the checkData check in packages/shared-ux/page/kibana_no_data/impl/src/kibana_no_data_page.tsx?

Reply:
No, it's to ensure the query coming in on the route is of type `esql` ... if it's not, we shouldn't route to the ES|QL view.

Reply:
In other words, I'm overriding the route-to-create-data-view behavior if the url contains an esql query.